tests/integration/specification_model_test.py

```diff
@@ -231,28 +231,29 @@ def test_model_name_conflict_messages() -> None:
                   end message;
             end Test;
         """,
-        r'^<stdin>:8:16: model: error: name conflict for type "Test::PDU"\n'
-        r'<stdin>:4:16: model: info: previous occurrence of "Test::PDU"$',
+        r'^<stdin>:8:21: model: error: name conflict for type "Test::PDU"\n'
+        r'<stdin>:4:21: model: info: previous occurrence of "Test::PDU"$',
     )
 
 
 def test_model_conflicting_refinements() -> None:
     assert_error_string(
         """
             package Test is
+               type X is null message;
                type PDU is
                   message
                      null
                         then Foo
                            with Size => 8;
                      Foo : Opaque;
                   end message;
-               for Test::PDU use (Foo => Test::PDU);
-               for PDU use (Foo => PDU);
+               for Test::PDU use (Foo => X);
+               for PDU use (Foo => X);
             end Test;
         """,
-        r'^<stdin>:11:16: model: error: conflicting refinement of "Test::PDU" with "Test::PDU"\n'
-        r"<stdin>:10:16: model: info: previous occurrence of refinement",
+        r'^<stdin>:12:16: model: error: conflicting refinement of "Test::PDU" with "Test::X"\n'
+        r"<stdin>:11:16: model: info: previous occurrence of refinement",
```

Comment:
What is the reason for introducing `X` here? While refining `PDU` with itself doesn't make much sense, refining `PDU` with a null message is wrong, although it is not detected due to #540. Furthermore, removing `Test::` in the refinement SDU removes a edge case.

Reply:
I don't remember why I changed it. Reverted in 2737b09b3ece6314b71a61d7ce152a30c41705f9.